### ISSUE TITLE
Designer review comments/feedback/todos for PhET-iO GAO "Model" screen

### ISSUE THREAD

**Issue by kathy-phet** [phetsims/gravity-and-orbits#293]
@arouinfar and @kathy-phet worked on the "Model" screen today, and have several comments/requests/questions to be considered by @samreid in advance of doing the overrides. 

- [x] We need clarity on how each screen's "activeProperty" works. What are the conditions underwhich it is "true"? Can we be a bit more clear in the Documentation too. 

UPDATE by @samreid and @chrisklus: We I updated the documentation (committed below), but we were uncertain whether it is clear enough or what more could be done.  It's difficult to experiment with `screen.activeProperty` in studio, because it always appears "true".  If you switch to a different screen, that screen immediately becomes active.  How do you recommend to improve further?

UPDATE 2/10/20: This is being addressed by changes to screenProperty. 

- [x] Discuss possibility of renaming gravityAndOrbits.modelScreen.model.gravityEnabledProperty to something the better matches the UI of the screen - like 
gravityAndOrbits.modelScreen.model.gravityOnOffProperty with values of "ON" and "OFF".
(we can live with it as designers if it doesn't sit well with the devs, but there is incoherence here between the UI and the PhET-iO Element name).

UPDATE by @samreid and @chrisklus: We implemented that suggestion so we could better understand its trade-offs.  A patch is below.  We believe this may be a step in the wrong direction because according to Model View Controller philosophy, the model should be developed independently from any view(s).  In this model, it is more straightforward to use a boolean, and @chrisklus and I agree that `A` below is superior to `B`
```js
// A
if (isGravityEnabled){...}

// B
if (gravityOnOffValue===ON){...}
```

UPDATE 2/10/20: @arouinfar and @kathy-phet are OK with leaving as boolean true/false here. The pattern is causing some issues in other sims (aka Balancing Act), so may need some dev + design discussion down the road to hone the pattern options a bit more. 

<details>

If we decide to pursue this further, we will need to evaluate where else this would have an impact.  For instance, in Sun we have OnOffSwitch which currently uses a boolean instead of an OnOffEnumeration.  Are there other places where we would want to change from true|false to ON|OFF and would we also need other enumerations in some cases, like YES|NO?

@jonathanolson added that we shouldn't create enumerations that are essentially true/false.  If necessary, we can add something in the downstream layer (in PhET-iO) that indicates "true" means "on" etc.

```diff
Index: js/common/model/GravityAndOrbitsModel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/GravityAndOrbitsModel.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/model/GravityAndOrbitsModel.js	(date 1576690324602)
@@ -21,6 +21,7 @@
   const EnumerationProperty = require( 'AXON/EnumerationProperty' );
   const gravityAndOrbits = require( 'GRAVITY_AND_ORBITS/gravityAndOrbits' );
   const NumberProperty = require( 'AXON/NumberProperty' );
+  const OnOffEnumeration = require( 'GRAVITY_AND_ORBITS/common/model/OnOffEnumeration' );
   const PhysicalConstants = require( 'PHET_CORE/PhysicalConstants' );
   const SpeedType = require( 'GRAVITY_AND_ORBITS/common/model/SpeedType' );
 
@@ -49,7 +50,7 @@
       this.isPlayingProperty = new BooleanProperty( false, { tandem: tandem.createTandem( 'isPlayingProperty' ) } );
       this.speedTypeProperty = new EnumerationProperty( SpeedType, SpeedType.NORMAL, { tandem: tandem.createTandem( 'speedTypeProperty' ) } );
 
-      this.gravityEnabledProperty = new BooleanProperty( true, { tandem: tandem.createTandem( 'gravityEnabledProperty' ) } );
+      this.gravityOnOffProperty = new EnumerationProperty( OnOffEnumeration, OnOffEnumeration.ON, { tandem: tandem.createTandem( 'gravityOnOffProperty' ) } );
       this.steppingProperty = new BooleanProperty( false );
       this.rewindingProperty = new BooleanProperty( false );
 
@@ -113,7 +114,7 @@
       this.isPlayingProperty.reset();
       this.speedTypeProperty.reset();
       this.showMeasuringTapeProperty.reset();
-      this.gravityEnabledProperty.reset();
+      this.gravityOnOffProperty.reset();
       this.steppingProperty.reset();
       this.rewindingProperty.reset();
       this.sceneIndexProperty.reset();
Index: js/phet-io/gravity-and-orbits-phet-io-elements-baseline.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/phet-io/gravity-and-orbits-phet-io-elements-baseline.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/phet-io/gravity-and-orbits-phet-io-elements-baseline.js	(date 1576690335641)
@@ -2121,7 +2121,7 @@
       "phetioTypeName": "PropertyIO<BooleanIO>"
     },
     "gravityAndOrbits.homeScreen.activeProperty": {
-      "phetioDocumentation": "Indicates whether the screen is active.  For single-screen simulations, the screen is always active.",
+      "phetioDocumentation": "Indicates whether the screen is currently displayed in the simulation.  For single-screen simulations, there is only one screen and it is always active.",
       "phetioDynamicElement": false,
       "phetioDynamicElementArchetype": false,
       "phetioEventType": "MODEL",
@@ -2875,7 +2875,7 @@
       "phetioTypeName": "PropertyIO<BooleanIO>"
     },
     "gravityAndOrbits.modelScreen.activeProperty": {
-      "phetioDocumentation": "Indicates whether the screen is active.  For single-screen simulations, the screen is always active.",
+      "phetioDocumentation": "Indicates whether the screen is currently displayed in the simulation.  For single-screen simulations, there is only one screen and it is always active.",
       "phetioDynamicElement": false,
       "phetioDynamicElementArchetype": false,
       "phetioEventType": "MODEL",
@@ -2887,7 +2887,7 @@
       "phetioStudioControl": true,
       "phetioTypeName": "PropertyIO<BooleanIO>"
     },
-    "gravityAndOrbits.modelScreen.model.gravityEnabledProperty": {
+    "gravityAndOrbits.modelScreen.model.gravityOnOffProperty": {
       "phetioDocumentation": "",
       "phetioDynamicElement": false,
       "phetioDynamicElementArchetype": false,
@@ -2898,7 +2898,7 @@
       "phetioReadOnly": false,
       "phetioState": true,
       "phetioStudioControl": true,
-      "phetioTypeName": "PropertyIO<BooleanIO>"
+      "phetioTypeName": "PropertyIO<EnumerationIO(ON|OFF)>"
     },
     "gravityAndOrbits.modelScreen.model.isPlayingProperty": {
       "phetioDocumentation": "",
@@ -17890,7 +17890,7 @@
       "phetioTypeName": "PropertyIO<BooleanIO>"
     },
     "gravityAndOrbits.toScaleScreen.activeProperty": {
-      "phetioDocumentation": "Indicates whether the screen is active.  For single-screen simulations, the screen is always active.",
+      "phetioDocumentation": "Indicates whether the screen is currently displayed in the simulation.  For single-screen simulations, there is only one screen and it is always active.",
       "phetioDynamicElement": false,
       "phetioDynamicElementArchetype": false,
       "phetioEventType": "MODEL",
@@ -17902,7 +17902,7 @@
       "phetioStudioControl": true,
       "phetioTypeName": "PropertyIO<BooleanIO>"
     },
-    "gravityAndOrbits.toScaleScreen.model.gravityEnabledProperty": {
+    "gravityAndOrbits.toScaleScreen.model.gravityOnOffProperty": {
       "phetioDocumentation": "",
       "phetioDynamicElement": false,
       "phetioDynamicElementArchetype": false,
@@ -17913,7 +17913,7 @@
       "phetioReadOnly": false,
       "phetioState": true,
       "phetioStudioControl": true,
-      "phetioTypeName": "PropertyIO<BooleanIO>"
+      "phetioTypeName": "PropertyIO<EnumerationIO(ON|OFF)>"
     },
     "gravityAndOrbits.toScaleScreen.model.isPlayingProperty": {
       "phetioDocumentation": "",
Index: js/common/model/OnOffEnumeration.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/OnOffEnumeration.js	(date 1576689801370)
+++ js/common/model/OnOffEnumeration.js	(date 1576689801370)
@@ -0,0 +1,19 @@
+// Copyright 2019, University of Colorado Boulder
+
+/**
+ * Enumeration that determines whether something is on or off.
+ *
+ * @author Sam Reid (PhET Interactive Simulations)
+ */
+define( require => {
+  'use strict';
+
+  // modules
+  const Enumeration = require( 'PHET_CORE/Enumeration' );
+  const gravityAndOrbits = require( 'GRAVITY_AND_ORBITS/gravityAndOrbits' );
+
+  return gravityAndOrbits.register( 'OnOffEnumeration', Enumeration.byKeys( [
+    'ON',
+    'OFF'
+  ] ) );
+} );
\ No newline at end of file
Index: js/phet-io/gravity-and-orbits-phet-io-types.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/phet-io/gravity-and-orbits-phet-io-types.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/phet-io/gravity-and-orbits-phet-io-types.js	(date 1576690335691)
@@ -333,6 +333,14 @@
       "supertype": "ActionIO<NullableIO<EventIO>>",
       "typeName": "EmitterIO<NullableIO<EventIO>>"
     },
+    "EnumerationIO(ON|OFF)": {
+      "documentation": "Possible values: ON,OFF.",
+      "events": [],
+      "methodOrder": [],
+      "methods": {},
+      "supertype": "ObjectIO",
+      "typeName": "EnumerationIO(ON|OFF)"
+    },
     "EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD)": {
       "documentation": "Possible values: SLOW_MOTION,NORMAL,FAST_FORWARD.",
       "events": [],
@@ -389,6 +397,19 @@
       "supertype": "ObjectIO",
       "typeName": "FunctionIO(BooleanIO,NullableIO<BooleanIO>)=>VoidIO"
     },
+    "FunctionIO(EnumerationIO(ON|OFF),NullableIO<EnumerationIO(ON|OFF)>)=>VoidIO": {
+      "documentation": "Wrapper for the built-in JS function type.<br><strong>Arguments:</strong> EnumerationIO(ON|OFF), NullableIO<EnumerationIO(ON|OFF)><br><strong>Return Type:</strong> VoidIO",
+      "events": [],
+      "methodOrder": [],
+      "methods": {},
+      "parameterTypes": [
+        "EnumerationIO(ON|OFF)",
+        "NullableIO<EnumerationIO(ON|OFF)>",
+        "VoidIO"
+      ],
+      "supertype": "ObjectIO",
+      "typeName": "FunctionIO(EnumerationIO(ON|OFF),NullableIO<EnumerationIO(ON|OFF)>)=>VoidIO"
+    },
     "FunctionIO(EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD),NullableIO<EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD)>)=>VoidIO": {
       "documentation": "Wrapper for the built-in JS function type.<br><strong>Arguments:</strong> EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD), NullableIO<EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD)><br><strong>Return Type:</strong> VoidIO",
       "events": [],
@@ -558,6 +579,17 @@
       "supertype": "ObjectIO",
       "typeName": "NullableIO<BooleanIO>"
     },
+    "NullableIO<EnumerationIO(ON|OFF)>": {
+      "documentation": "A wrapper to wrap another IOType, adding support for null.",
+      "events": [],
+      "methodOrder": [],
+      "methods": {},
+      "parameterTypes": [
+        "EnumerationIO(ON|OFF)"
+      ],
+      "supertype": "ObjectIO",
+      "typeName": "NullableIO<EnumerationIO(ON|OFF)>"
+    },
     "NullableIO<EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD)>": {
       "documentation": "A wrapper to wrap another IOType, adding support for null.",
       "events": [],
@@ -1008,6 +1040,50 @@
       "supertype": "ObjectIO",
       "typeName": "PropertyIO<BooleanIO>"
     },
+    "PropertyIO<EnumerationIO(ON|OFF)>": {
+      "documentation": "Observable values that send out notifications when the value changes. This differs from the traditional listener pattern in that added listeners also receive a callback with the current value when the listeners are registered. This is a widely-used pattern in PhET-iO simulations.",
+      "events": [
+        "changed"
+      ],
+      "methodOrder": [
+        "link",
+        "lazyLink"
+      ],
+      "methods": {
+        "getValue": {
+          "documentation": "Gets the current value.",
+          "parameterTypes": [],
+          "returnType": "EnumerationIO(ON|OFF)"
+        },
+        "lazyLink": {
+          "documentation": "Adds a listener which will be called when the value changes. This method is like \"link\", but without the current-value callback on registration. The listener takes two arguments, the new value and the previous value.",
+          "parameterTypes": [
+            "FunctionIO(EnumerationIO(ON|OFF),NullableIO<EnumerationIO(ON|OFF)>)=>VoidIO"
+          ],
+          "returnType": "VoidIO"
+        },
+        "link": {
+          "documentation": "Adds a listener which will be called when the value changes. On registration, the listener is also called with the current value. The listener takes two arguments, the new value and the previous value.",
+          "parameterTypes": [
+            "FunctionIO(EnumerationIO(ON|OFF),NullableIO<EnumerationIO(ON|OFF)>)=>VoidIO"
+          ],
+          "returnType": "VoidIO"
+        },
+        "setValue": {
+          "documentation": "Sets the value of the Property. If the value differs from the previous value, listeners are notified with the new value.",
+          "invocableForReadOnlyElements": false,
+          "parameterTypes": [
+            "EnumerationIO(ON|OFF)"
+          ],
+          "returnType": "VoidIO"
+        }
+      },
+      "parameterTypes": [
+        "EnumerationIO(ON|OFF)"
+      ],
+      "supertype": "ObjectIO",
+      "typeName": "PropertyIO<EnumerationIO(ON|OFF)>"
+    },
     "PropertyIO<EnumerationIO(SLOW_MOTION|NORMAL|FAST_FORWARD)>": {
       "documentation": "Observable values that send out notifications when the value changes. This differs from the traditional listener pattern in that added listeners also receive a callback with the current value when the listeners are registered. This is a widely-used pattern in PhET-iO simulations.",
       "events": [
Index: js/common/view/GravityAndOrbitsControlPanel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/view/GravityAndOrbitsControlPanel.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/view/GravityAndOrbitsControlPanel.js	(date 1576690324592)
@@ -44,7 +44,7 @@
       const sections = [
         new SceneSelectionControls( model.sceneIndexProperty, model.getScenes(), merge( { tandem: options.tandem.createTandem( 'sceneControl' ) }, MENU_SECTION_OPTIONS ) ),
         makeTopSeparatorRectangle(),
-        new GravityControl( model.gravityEnabledProperty, merge( { tandem: options.tandem.createTandem( 'gravityControl' ) }, MENU_SECTION_OPTIONS ) ),
+        new GravityControl( model.gravityOnOffProperty, merge( { tandem: options.tandem.createTandem( 'gravityControl' ) }, MENU_SECTION_OPTIONS ) ),
         makeBottomSeparatorRectangle(),
         new CheckboxPanel( model, merge( { tandem: options.tandem.createTandem( 'checkboxPanel' ) }, MENU_SECTION_OPTIONS ) )
       ];
Index: js/common/model/GravityAndOrbitsPhysicsEngine.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/GravityAndOrbitsPhysicsEngine.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/model/GravityAndOrbitsPhysicsEngine.js	(date 1576690324585)
@@ -33,12 +33,12 @@
 
     /**
      * @param {GravityAndOrbitsClock} clock
-     * @param {Property.<boolean>} gravityEnabledProperty flag to indicate whether gravity is on or off.
+     * @param {Property.<OnOffEnumeration>} gravityOnOffProperty flag to indicate whether gravity is on or off.
      */
-    constructor( clock, gravityEnabledProperty ) {
+    constructor( clock, gravityOnOffProperty ) {
 
       // @private
-      this.gravityEnabledProperty = gravityEnabledProperty;
+      this.gravityOnOffProperty = gravityOnOffProperty;
 
       this.clock = clock; // @public
       this.bodies = []; // @public - contains the sun, moon, earth, satellite
@@ -53,7 +53,7 @@
       } );
 
       // Have to update force vectors when gravity gets toggled on and off, otherwise displayed value won't update
-      this.gravityEnabledProperty.link( this.updateForceVectors.bind( this ) );
+      this.gravityOnOffProperty.link( this.updateForceVectors.bind( this ) );
     }
 
     /**
@@ -99,7 +99,7 @@
 
       // Compute the next state for each body based on the current state of all bodies in the system.
       const bodyStates = this.bodies.map( getBodyState );
-      const newState = new ModelState( bodyStates, this.clock ).getNextState( dt, this.gravityEnabledProperty );
+      const newState = new ModelState( bodyStates, this.clock ).getNextState( dt, this.gravityOnOffProperty );
 
       // Set each body to its computed next state.
       // assumes that ModelState.getBodyState returns states in the same order as the container (ArrayList) used for
Index: js/common/GravityAndOrbitsScene.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/GravityAndOrbitsScene.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/GravityAndOrbitsScene.js	(date 1576690324608)
@@ -99,7 +99,7 @@
 
       // @private
       const clock = new GravityAndOrbitsClock( dt, model.steppingProperty, this.speedTypeProperty, tandem.createTandem( 'clock' ) );
-      this.physicsEngine = new GravityAndOrbitsPhysicsEngine( clock, model.gravityEnabledProperty );
+      this.physicsEngine = new GravityAndOrbitsPhysicsEngine( clock, model.gravityOnOffProperty );
 
       Property.multilink( [ model.isPlayingProperty, this.activeProperty ], ( playButtonPressed, active ) =>
         this.physicsEngine.clock.setRunning( playButtonPressed && active )
Index: js/common/view/GravityControl.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/view/GravityControl.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/view/GravityControl.js	(date 1576690324614)
@@ -16,6 +16,7 @@
   const HBox = require( 'SCENERY/nodes/HBox' );
   const merge = require( 'PHET_CORE/merge' );
   const Node = require( 'SCENERY/nodes/Node' );
+  const OnOffEnumeration = require( 'GRAVITY_AND_ORBITS/common/model/OnOffEnumeration' );
   const PhetFont = require( 'SCENERY_PHET/PhetFont' );
   const Text = require( 'SCENERY/nodes/Text' );
 
@@ -31,10 +32,10 @@
   class GravityControl extends Node {
 
     /**
-     * @param {Property.<boolean>} gravityEnabledProperty
+     * @param {Property.<OnOffEnumeration>} gravityOnOffProperty
      * @param {Object} [options] - This object contains options for main node of gravity menu.
      */
-    constructor( gravityEnabledProperty, options ) {
+    constructor( gravityOnOffProperty, options ) {
       super( options );
 
       const gravityTextNode = new Text( gravityString, TEXT_OPTIONS );
@@ -44,8 +45,8 @@
       this.addChild( new HBox( {
         spacing: 10, bottom: 2, resize: false, children: [
           gravityTextNode,
-          new AquaRadioButton( gravityEnabledProperty, true, onTextNode, merge( { tandem: options.tandem.createTandem( 'gravityOnRadioButton' ) }, RADIO_OPTIONS ) ),
-          new AquaRadioButton( gravityEnabledProperty, false, offTextNode, merge( { tandem: options.tandem.createTandem( 'gravityOffRadioButton' ) }, RADIO_OPTIONS ) )
+          new AquaRadioButton( gravityOnOffProperty, OnOffEnumeration.ON, onTextNode, merge( { tandem: options.tandem.createTandem( 'gravityOnRadioButton' ) }, RADIO_OPTIONS ) ),
+          new AquaRadioButton( gravityOnOffProperty, OnOffEnumeration.OFF, offTextNode, merge( { tandem: options.tandem.createTandem( 'gravityOffRadioButton' ) }, RADIO_OPTIONS ) )
         ]
       } ) );
     }
Index: js/common/model/ModelState.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/ModelState.js	(revision 630a621d14e7567a03fa2c583175881efb8349f7)
+++ js/common/model/ModelState.js	(date 1576690324580)
@@ -14,6 +14,7 @@
 
   // modules
   const gravityAndOrbits = require( 'GRAVITY_AND_ORBITS/gravityAndOrbits' );
+  const OnOffEnumeration = require( 'GRAVITY_AND_ORBITS/common/model/OnOffEnumeration' );
   const PhysicalConstants = require( 'PHET_CORE/PhysicalConstants' );
   const Vector2 = require( 'DOT/Vector2' );
 
@@ -43,12 +44,12 @@
      *
      * @public
      * @param {number} dt
-     * @param {Property.<boolean>} gravityEnabledProperty
+     * @param {Property.<boolean>} gravityOnOffProperty
      * @returns {ModelState}
      */
-    getNextState( dt, gravityEnabledProperty ) {
+    getNextState( dt, gravityOnOffProperty ) {
 
-      if ( gravityEnabledProperty.get() ) {
+      if ( gravityOnOffProperty.get() === OnOffEnumeration.ON ) {
         return this.getNextInteractingState( dt );
       }
       else {

```

</details>

- [x] Can gravityAndOrbits.modelScreen.model.sceneIndexProperty actually be an enumeration property with the same scene names that match the other tandem names? Right now you have to know the mapping of 0, 1, 2, in order to understand this property and datastream events. (Also disable the studio Slider for this element. Using it breaks Studio)

UPDATE from @samreid @chrisklus and @jonathanolson: We revised this in the commit below, thanks for the good suggestion!  We cut out an unused intermediate number property.

- [x] Many of the model properties needs "Documentation" and "Units": forceProperty, massProperty, positionProperty, velocityProperty, rotationProperty, diameter, etc. 

- [x] gravityAndOrbits.modelScreen.model.planetMoonScene.moon.isCollidedProperty seems like it is critical it should be "read only". What was the decided on policy for this? Is it in the code or in the overrides? 

UPDATE from @samreid: this should be done in the code, since there is a single central point for all of these Properties.  I'll commit this change shortly.

- [X] (see #300) We do not understand what the "isRunningProperty" is in clock (e.g. gravityAndOrbits.modelScreen.model.planetMoonScene.clock.isRunningProperty). It doesn't seem to do anything? Or to be impacted by play/pause?

UPDATE from @samreid:  The code that sets that property is:

```js
      Property.multilink( [ model.isPlayingProperty, this.activeProperty ], ( playButtonPressed, active ) =>
        this.physicsEngine.clock.setRunning( playButtonPressed && active )
      );
```

That is, the scene's clock is running if the entire model is playing and the scene is active.  I see that it is affected by the play/pause button, but only if you are on the right scene.

- [X] (see #298) We also do not understand what the "isMoveableProperty" is for (e.g. gravityAndOrbits.modelScreen.model.planetMoonScene.moon.isMovableProperty). When we click it, the moon stops orbiting and you see it sort of repelled which is really odd? Please help us understand why this is in the code. 

UPDATE from @samreid: The "model" screen Sun is fixed.  I saw this code comment:

```js
      // Sun shouldn't move in model modes
      this.sun.isMovable = false;
```

How would you like to deal with this in studio?

UPDATE 2/10/20: Looks like there is still something odd going on with isMovableProperty, see #298 

- [x] The following 2 tandems should not appear in the "Model" screen: 
gravityAndOrbits.modelScreen.model.showMassProperty 
gravityAndOrbits.modelScreen.model.showMeasuringTapeProperty

- [x] For this tandem gravityAndOrbits.modelScreen.model.planetMoonScene.zoomLevelProperty using the Studio slider breaks studio. Can you remove the slider from Studio

UPDATE from @samreid: I fixed the range for the zoom Property and it's working nicely in studio now.  I didn't remove it yet, but can do so if preferred.

- [x] For saving data out of the simulation, we would like to create several new properties that would be natural data that you would want as an instructional designer having students save or plot data: 
1. a "forceMagnitudeProperty" for each body and perhaps a "forceDirectionProperty" (in degrees where 0 is horizontal to the right and conventions as in "Vector Addition")
2. a "velocityMagnitudeProperty" for each body and perhaps a "velocityDirectionProperty"
3. a "planetMoonDistanceProperty" at the scene level for each pair of bodies in that scene. e.g. 
gravityAndOrbits.modelScreen.model.planetMoonScene.planetMoonDistanceProperty
We understand these can be "calculated" from properties that already exist, but since these are the data you would want to have students collect as an instructional designer I think we should do the work in the sim and not ask each client to do it in the wrappers.

I'll move this to a new issue: https://github.com/phetsims/gravity-and-orbits/issues/295

- [x] For these kinds of tandems (individual scene mass control panel visibility) gravityAndOrbits.modelScreen.view.massesControlPanel.planetMoonSceneMassesControlPanel.visibleProperty
we feel they need to be "read only" because you can really get in a pickle if you set them since they are set and controlled by the sim itself. Again - do we do this with overrides, or is this in the code since it basically breaks the code if it is not done. Or is it done in overrides - but the dev is supposed to do these "need to be read only" cases there. I just cannot recall what was decided on this piece of setting the metadata. 

UPDATE from @samreid: I believe @pixelzoom advocated that changes like these should be done in the code when possible, so it can be properly documented and maintained, and because it would have buggy behavior if not done correctly.

- [x] We don't understand the set of ids of this nature
gravityAndOrbits.modelScreen.view.massesControlPanel.starPlanetSceneMassesControlPanel.planetMassControl.enabledRangeProperty
Can you please explain to us and add "Documentation"? 

UPDATE by @samreid: I added this documentation:

> Sliders support two ranges: the outer range which specifies the min and max of the track and the enabledRangeProperty, which determines how low and high the thumb can be dragged within the track.

- [x] Question re the zoom control. From an instructional design, we will just feature at the highest level
gravityAndOrbits.modelScreen.view.playAreaNode.starPlanetSceneView.zoomControl.visibleProperty
Is the reason all the pieces parts need to be instrumented is for playback? And event data streams?

UPDATE by @samreid: Playback would work without instrumenting these subcomponents since it uses low-level scenery events, but we would like to instrument them for their event data streams.  Perhaps the solution is to un-feature those subcomponents?

- [X] (see #301) The measuringTapeCheckbox is instrumented, but the MeasuringTapeNode itself does not appear to be instrumented.

NOTE: Many of the comments will also pertain to the toScaleScreeen, so apply as many that are universal to both as possible. 

@arouinfar - Please edit if I missed anything.

**Comment by arouinfar** [comment]
@kathy-phet I read through my notes, and I couldn't find anything that you hadn't already documented in the issue.

**Comment by arouinfar** [comment]
Comparing notes to Projectile Motion, I just realized that the MeasuringTapeNode doesn't appear to be instrumented (just the checkbox to toggle it). I've edited @kathy-phet's original comment to add this item.

**Comment by samreid** [comment]
For the preceding commit, I was unsure whether to go with readOnly: true or studioControl: false.  @zepumph can you please comment?

**Comment by pixelzoom** [comment]
@samreid update in https://github.com/phetsims/gravity-and-orbits/issues/293#issue-533171338:
> UPDATE from @samreid: I believe @pixelzoom advocated that changes like these should be done in the code when possible, so it can be properly documented and maintained, and because it would have buggy behavior if not done correctly.

That's correct.  For sim-specific elements,  I think it's desirable to move most metadata changes back into the code. The exception is `phetioFeatured`

For common-code elements, metadata changes should be reviewed to see whether they are appropriate to move back into the code.   For example, the `phetioDocumentation` change for screen.activeProperty` was presumably handle in code (not in overrides.js) after verifying that it was an appropriate change for all sims.

**Comment by zepumph** [comment]
>For the preceding commit, I was unsure whether to go with readOnly: true or studioControl: false. @zepumph can you please comment?

It looks like it is very buggy to every set that visibility. I felt this especially after setting it to readOnly false, and toggling it in studio. From this I recommend `phetioReadOnly`, as I wouldn't want any way to toggle that, since the sim will handle that for us. 

I also grunt updated. Back to you!

**Comment by samreid** [comment]
We addressed or commented on each of the requested items.  @kathy-phet and/or @arouinfar can you please review?  For any nontrivial issues that are not complete, I recommend to open a new issue.  Also, my notes indicate we would like to publish this "mid-February" and this is one of the last issues that needs to be completed before RC testing.  We could also schedule a design meeting or a side meeting if this helps us reach our goal on time.

**Comment by kathy-phet** [comment]
- [ ] Regarding, isRunningProperty and isPlayingProperty, we would like to have documentation on both of these. Suggestion: 
isRunningProperty = "This value is true when [INSERT SCENE NAME] is the active scene AND the sim is playing (isPlayingProperty is 'true')."
isPlayingProperty = "This value is true [SCREEN] is the active screen AND the play/pause button is in play mode."
NOTE: We are a bit confused about how these Properties should change when you change screens. There do not seem to be any messages logged to the console when you change screens?

**Comment by pixelzoom** [comment]
@kathy-phet Regarding your descriptions in the previous comment.... 
> isRunningProperty = "This value is true when [INSERT SCENE NAME] is the active scene AND the sim is playing (isPlayingProperty is 'true')."
> isPlayingProperty = "This value is true [SCREEN] is the active screen AND the play/pause button is in play mode."

I just examined GAO in the debugger.  I see no change to isPlayingProperty when a screen becomes inactive.  And isRunningProperty is also therefore unaffected by whether the screen is active.

**Comment by pixelzoom** [comment]
Here's what I see in GAO for these 2 Properties:

* `isPlayingProperty` applies to a screen's model, and is true if the play/pause button for the screen is in the "play" mode.

* `isRunningProperty` applies to a scene, and is true if the scene is selected and isPlayingProperty is true for the model that the scene belongs to.

Switching between screens results in no change to either Property.

**Comment by samreid** [comment]
I'll be moving the preceding 3 comments to a new issue.

**Comment by arouinfar** [comment]
@samreid I don't believe there is anything in this issue that is still awaiting designer review, so I'm going to unassign @kathy-phet and myself.

**Comment by samreid** [comment]
@kathy-phet and/or @arouinfar can this issue be closed?  If there is remaining work to be done, can it be in a new issue?

**Comment by arouinfar** [comment]
@samreid the unchecked items are all being tracked in separate issues. I've added references to those issues and checked them off. We can go ahead and close this behemoth.
